sound/soc/sof/intel/hda-dsp.c

```diff
@@ -375,6 +384,14 @@ static int hda_resume(struct snd_sof_dev *sdev)
 	snd_hdac_ext_bus_ppcap_enable(bus, true);
 	snd_hdac_ext_bus_ppcap_int_enable(bus, true);
 
+	/* check jack status based on controller status */
+	hda_codec_jack_poll(sdev, status);
```

Comment:
@RanderWang the legacy hda driver seems to do this only for runtime resume. Should we also have a check here for runtime resume and do this only then?

Reply:
yes, I will refine it. Thanks!